### ISSUE TITLE
`gitnr search` view reacts to both `Press` and `Release` events

### ISSUE THREAD

**Issue by supleed2** [reemus-dev/gitnr#3]
In `fn handle_key_events` ([here](https://github.com/reemus-dev/gitnr/blob/main/src/commands/search/handlers.rs#L15)), the `event.kind` is not checked with keyboard input.

I think this event is created when pressing and releasing a key, so every key is reacted to twice. Trying to select using enter selects and then unselects, or using arrow keys moves 2 items at a time.

This is on Windows, but looking at the [crossterm docs](https://docs.rs/crossterm/latest/crossterm/event/struct.KeyEvent.html), the event kind needs to be opted-in for unix systems so it might not happen for you.

**Comment by realfresh** [comment]
Hey, thanks for the heads-up and providing the relevant docs. 

Didn't notice that at all. Thankfully, I think the fix was quite simple (https://github.com/reemus-dev/gitnr/commit/7dffa5b63f5b1812711eee4302c0e493ba0f903a). 

I also enabled `event.kind` for Unix systems as well, so things are consistent across platforms.

The new release [v0.1.1](https://github.com/reemus-dev/gitnr/releases/tag/v0.1.1) has that fix. 

Let me know if it works fine, I don't use Windows and as such can't test it out easily.

**Comment by supleed2** [comment]
Hey, thanks for reacting so quickly. The new version actually doesn't work now hah. Screenshot below:

![image](https://github.com/reemus-dev/gitnr/assets/21363892/1b41dc6a-7e80-4cba-acac-e7a8f29681bb)

I think you might need to gate the `push` and `pop` of `KeyboardEnhancementFlags`, maybe behind cfg(not(windows)), I'm not sure on the syntax. But the rest of the commit looks correct. This is probably because, as the crossterm docs say, it's always enabled for Windows and the enhancement flags don't exist.

PS: would you consider publishing to crates.io? It would allow for installation via cargo-binstall too, since it pulls metadata from there to then locate the GitHub release url.

**Comment by realfresh** [comment]
My bad mate...lol. Thanks for showing the error and pointing me in the right direction.

Managed to fix that issue and tested it in my Windows VM. Also managed to fix a few other issues too.

Release [v0.1.3](https://github.com/reemus-dev/gitnr/releases/tag/v0.1.3) is available now.

I've also set up the GH action to publish the crate, so it's available on [crates.io](https://crates.io/crates/gitnr) now. I've also configured it to work with `cargo-binstall` as well. Check out the README for more info on the new install methods.
